composeApp/src/commonMain/kotlin/org/ooni/probe/domain/GetStorageUsed.kt

```diff
@@ -0,0 +1,35 @@
+package org.ooni.probe.domain
+
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+import okio.FileSystem
+import okio.Path.Companion.toPath
+
+class GetStorageUsed(
+    private val baseFileDir: String,
+    private val fileSystem: FileSystem,
+    private val cacheDir: String,
+) {
+    private val storageUsed = MutableStateFlow<Long>(0)
+
+    fun observe(): StateFlow<Long> {
+        if (storageUsed.value == 0L) {
+            update()
```

Comment:
Since this should become a `suspend`, it's better if we run it as:

```
storageUsed.asStateFlow()
  .onStart { if (storageUsed.value == 0L) update() }
```